components/cdc/src/endpoint.rs

```diff
@@ -607,6 +613,19 @@ impl<T: 'static + RaftStoreRouter<E>, E: KvEngine> Endpoint<T, E> {
             return;
         }
 
+        if (kv_api == ChangeDataRequestKvApi::RawKv && api_version != ApiVersion::V2)
+            || (kv_api == ChangeDataRequestKvApi::TxnKv)
```

Comment:
Could you add the check to `FeatureGate`, like `FeatureGate::validate_kv_api()`?